addons/website_sale/controllers/main.py

```diff
@@ -1179,7 +1179,9 @@ def shop_address_submit(
         """
         order_sudo = request.website.sale_get_order()
         if redirection := self._check_cart(order_sudo):
-            return redirection
+            # Return the redirect url in a json object as the caller expects a json object, not an
+            # html document. The client will then do the redirection manually.
```

Comment:
The comment is 100% correct, but a bit overkill IMHO, as you can easily guess why we're doing it. See the main return value at the bottom of the method for example.
```suggestion
```